go-controller/pkg/factory/factory.go

```diff
@@ -390,10 +406,14 @@ type WatchFactory struct {
 	// requirements with atomic accesses
 	handlerCounter uint64
 
-	iFactory  informerfactory.SharedInformerFactory
-	informers map[reflect.Type]*informer
+	iFactory    informerfactory.SharedInformerFactory
+	efFactory   egressfirewallinformerfactory.SharedInformerFactory
+	efClientset egressfirewallclientset.Interface
+	crdFactory  apiextensionsinformerfactory.SharedInformerFactory
+	informers   map[reflect.Type]*informer
 
-	stopChan chan struct{}
+	stopChan               chan struct{}
+	egressFirewallStopChan chan struct{}
```

Comment:
do we need a `crdFactoryStopChan` for the `crdFactory`?

Reply:
unless we want to specifically stop the CRD watcher separately from the rest of the informers we do not need it. I don't think we want/need the ability to stop watching CRDs

Reply:
I don't think we need to stop it separately; let's just use the same stop channel for everything.